contracts/TransferLimitModule.sol

```diff
@@ -0,0 +1,445 @@
+pragma solidity ^0.5.0;
+
+import "../base/Module.sol";
+import "../base/ModuleManager.sol";
+import "../base/OwnerManager.sol";
+import "../common/Enum.sol";
+import "../common/SignatureDecoder.sol";
+import "../common/SecuredTokenTransfer.sol";
+import "../external/DutchExchangeInterface.sol";
+import "../external/PriceOracleInterface.sol";
+import "../external/SafeMath.sol";
+
+
+/// @title Transfer Limit Module - Allows to transfer limited amounts of ERC20 tokens and Ether.
+contract TransferLimitModule is Module, SignatureDecoder, SecuredTokenTransfer {
+    using SafeMath for uint256;
+
+    string public constant NAME = "Transfer Limit Module";
+    string public constant VERSION = "0.0.1";
+
+    // keccak256(
+    //     "EIP712Domain(address verifyingContract)"
+    // );
+    bytes32 public constant DOMAIN_SEPARATOR_TYPEHASH = 0x035aff83d86937d35b32e04f0ddc6ff469290eef2f1b692d8a815c89404d4749;
+
+    // keccak256(
+    //     "TransferLimitTx(address token,address to,uint256 amount,uint256 refund,address gasToken,address refundReceiver,uint256 nonce)"
+    // );
+    bytes32 public constant TX_TYPEHASH = 0x7e238b3eff6b836db6c4c9737afec4334a0d6dd13c3bbba4407f3e8506784733;
+
+    event TransferFailed(bytes32 txHash);
+
+    bytes32 public domainSeparator;
+
+    // transferLimits mapping maps token address to transfer limit settings.
+    mapping (address => TransferLimit) public transferLimits;
+
+    // Time period for which the transfer limits apply, in seconds.
+    uint256 public timePeriod;
+
+    // Start of the time period, during which the last transfer occured (common for all tokens).
+    uint256 public lastStartTime;
+
+    // If true, the expenditure between [now - timePeriod, now] will be considered.
+    bool public rolling;
+
+    // Global limit on all transfers, specified in Wei.
+    uint256 public globalWeiCap;
+
+    // Total amount of Wei spent in current time period.
+    uint256 public totalWeiSpent;
+
+    // Global limit on transfers, specified in usd (dai).
+    uint256 public globalDaiCap;
+
+    // Total amount of dai spent in current time period.
+    uint256 public totalDaiSpent;
+
+    // Number of required confirmations for a transfer.
+    uint256 public threshold;
```

Comment:
This threshold should probably be updatable by the safe, since it is possible that the safe owner counts changes after the setup of the module and then this module cannot be used